### PR TITLE
Automatic and manual alignment options for dp_bands

### PR DESCRIPTION
**Pull request by vanderhe**
Adds automatic and manual alignment options to the dp_bands script of the dptools.

I have found this feature to be extremely convenient, so decided to add it to the official dptools.

### CODE REVIEW

**Comment by vanderhe** [comment]
> The integer case should be half way between the VBM and CBM, not at the VBM (or add another flag to specifically align at VBM).

I would disagree, provided my use cases somehow reflect the "common" requirements of users, as in 99% of my applications I like to align my semiconductor/insulator band structures at the VBM, not the Fermi level. That is why I called it "auto-align", not something more specific like "align-at-fermi" or so. But that might just be personal preference, let's maybe discuss next devel. meeting. Meanwhile I implement both functionalities and we then decide how to invoke each of them from the users' perspective.

> Can you make the test data a bit smaller, or is there a reason for so many k-points?

Of course, no problem.

File: tools/dptools/src/dptools/scripts/dp_bands.py [hunk 1]
```diff
@@ -40,15 +40,34 @@ def parse_cmdline_args(cmdlineargs=None):
...
     parser.add_argument("-N", "--no-enumeration", action="store_false",
                         default=True, dest="enum", help=msg)
+
     msg = "create separate band structure for each spin channel"
     parser.add_argument("-s", "--separate-spins", action="store_true",
                         default=False, dest="spinsep", help=msg)
+
+    msg = "aligns an estimate of the level of half-occupation at zero " +\
+        "energy\ninteger occupations: Shifts the VBM to zero energy.\n" +\
+        "fractional occupations: Shifts Fermi-type level, i.e. the highest" +\
```
Comment (bhourahine) [inline tools/dptools/src/dptools/scripts/dp_bands.py:56]:
```suggestion
        "fractional occupations: Shifts approximate Fermi level, i.e. the highest" +\
```

**Review by bhourahine — APPROVED** [review]
Couple of minor suggestions, but good to go.

File: tools/dptools/src/dptools/scripts/dp_bands.py [hunk 2]
```diff
@@ -40,15 +40,34 @@ def parse_cmdline_args(cmdlineargs=None):
...
+
     msg = "do not use the first column of the output to enumerate the k-points"
     parser.add_argument("-N", "--no-enumeration", action="store_false",
                         default=True, dest="enum", help=msg)
+
     msg = "create separate band structure for each spin channel"
     parser.add_argument("-s", "--separate-spins", action="store_true",
                         default=False, dest="spinsep", help=msg)
+
+    msg = "aligns an estimate of the level of half-occupation at zero " +\
```
[the remark this answers is not in the document]
Comment (vanderhe) [reply tools/dptools/src/dptools/scripts/dp_bands.py]: Thanks Ben.

**Pull request merged**